state/watcher/watcher.go

```diff
@@ -441,7 +464,15 @@ func (w *Watcher) sync() error {
 		}
 	}
 	if err := iter.Close(); err != nil {
```

Comment:
Isn't there an iter.Err() that we should be looking for, rather than just using Close?